modules/fedimint-lnv2-client/src/lib.rs

```diff
@@ -0,0 +1,725 @@
+
+        let contract_amount = payment_info
+            .payment_fees
+            .outgoing_contract_amount(invoice_msats);
+
+        let contract = OutgoingContract {
+            payment_hash: *invoice.payment_hash(),
+            amount: contract_amount,
+            expiration: consensus_block_count + expiration_delta,
+            claim_pk: payment_info.public_key,
+            refund_pk: refund_keypair.public_key(),
+            ephemeral_pk,
+            invoice_hash: invoice.consensus_hash(),
+        };
+
+        let operation_id = OperationId(contract.consensus_hash::<sha256::Hash>().into_inner());
+
+        let client_output = ClientOutput::<LightningOutput, LightningClientStateMachines> {
+            output: LightningOutput::Outgoing(contract.clone()),
+            state_machines: Arc::new(move |txid, _| {
+                vec![LightningClientStateMachines::Send(SendStateMachine {
+                    operation_id,
+                    contract: contract.clone(),
+                    state: SendSMState::Funding(send_sm::Funding {
+                        txid,
+                        gateway_api: gateway_api.clone(),
+                        invoice: invoice.clone(),
+                        refund_keypair,
+                    }),
+                })]
+            }),
+        };
+
+        let client_output = self.client_ctx.make_client_output(client_output);
+        let transaction = TransactionBuilder::new().with_output(client_output);
+
+        self.client_ctx
+            .finalize_and_submit_transaction(
+                operation_id,
+                LightningCommonInit::KIND.as_str(),
+                |_, _| LightningOperationMeta,
+                transaction,
+            )
+            .await?;
+
+        Ok(operation_id)
+    }
+
+    pub async fn subscribe_send(
+        &self,
+        operation_id: OperationId,
+    ) -> anyhow::Result<UpdateStreamOrOutcome<SendState>> {
+        let operation = self.client_ctx.get_operation(operation_id).await?;
+        let mut stream = self.notifier.subscribe(operation_id).await;
+        let client_ctx = self.client_ctx.clone();
+        let module_api = self.module_api.clone();
+
+        Ok(operation.outcome_or_updates(&self.client_ctx.global_db(), operation_id, move || {
+            stream! {
+                loop {
+                    if let Some(LightningClientStateMachines::Send(state)) = stream.next().await {
+                        match state.state {
+                            SendSMState::Funding(..) => yield SendState::Funding,
+                            SendSMState::Funded(..) => yield SendState::Funded,
+                            SendSMState::Success(preimage) => {
+                                // the preimage has been verified by the state machine previously
+                                assert!(state.contract.verify_preimage(&preimage));
+
+                                yield SendState::Success(preimage);
```

Comment:
In legacy send, we wait for change since we might be paying an invoice with ecash notes that we can't make an exact amount of. Do we need to do that here too?

Reply:
@joschisan would be good to address this before merging